internal/multicluster/internal/types/exported_services.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package types
+
+import (
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/internal/resource"
+	pbmulticluster "github.com/hashicorp/consul/proto-public/pbmulticluster/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+func RegisterExportedServices(r resource.Registry) {
+	r.Register(resource.Registration{
+		Type:     pbmulticluster.ExportedServicesType,
+		Proto:    &pbmulticluster.ExportedServices{},
+		Scope:    resource.ScopeNamespace,
+		Mutate:   MutateExportedServices,
+		Validate: ValidateExportedServices,
+		ACLs: &resource.ACLHooks{
+			Read:  aclReadHookExportedServices,
+			Write: aclWriteHookExportedServices,
+			List:  aclListHookExportedServices,
+		},
+	})
+}
+
+func aclReadHookExportedServices(authorizer acl.Authorizer, authzContext *acl.AuthorizerContext, _ *pbresource.ID, res *pbresource.Resource) error {
+	var exportedService pbmulticluster.ExportedServices
```

Comment:
🚗 drive-by comment: You need a nil check on `res` for read ACL hooks because resource service will not pass in the resource object unless you return `resource.ErrNeedResource`. 

https://github.com/hashicorp/consul/blob/main/agent/grpc-external/services/resource/read.go#L53C1-L53C48

Here is an example: 

https://github.com/hashicorp/consul/blob/d52ee6a2224ed340d8697db6cd75f0b6dda73475/internal/catalog/internal/types/health_status.go#L77-L79

Reply:
Thanks